addon/services/media.js

```diff
@@ -77,35 +77,48 @@ import Evented from '@ember/object/evented';
 * @class     Media
 * @extends   Ember.Object
 */
-export default Service.extend(Evented, {
+export default class MediaService extends Service.extend(Evented) {
+  // Ember only sets Ember.testing when tests are starting
   // eslint-disable-next-line ember/no-ember-testing-in-module-scope
-  _mocked: Ember.testing,
-  _mockedBreakpoint: 'desktop',
+  _mocked = Ember.testing;
+  _mockedBreakpoint = 'desktop';
+
+  /**
+  * @property  _matches
+  * @type      Array
+  */
+  @tracked _matches;
+
   /**
   * A set of matching matchers.
   *
   * @property  matches
-  * @type      Ember.NativeArray
-  * @default   Ember.NativeArray
+  * @type      Array
   */
-  matches: computed('_mocked', '_mockedBreakpoint', function() {
-    return A(this.get('_mocked') ? [this.get('_mockedBreakpoint')] : []);
-  }),
+  get matches() {
+    if (this._matches) {
+      return this._matches
+    }
+
+    return (Ember.testing && this._mocked) ? [this._mockedBreakpoint] : [];
+  }
+
+  set matches(value) {
+    this._matches = value;
+  }
```

Comment:
These getters/setters + `@tracked _matches`.

First, `matches` can't be set in init and needs to be asked for lazily.  This is b/c `setBreakpoints` is run after init.  If a user has `setBreakpoints('auto')`, then it will still have the default testing `desktop` value.

Second, `Ember.testing && this._mocked` is necessary due to the change the you should never assign Ember.testing to a variable anymore.  `this._mocked` is used to turn off the condition in testing if `setBreakpoints('auto')` is set by the user.